src/test_linked_list.py

```diff
@@ -0,0 +1,127 @@
+"""Testing the linked_list class - CF 401 Python Week 2 Assignment."""
+import pytest
+
+PARAMS_SAMPLE_LIST = ["something", 1, "pear", 3, "apple"]
+
+@pytest.fixture
+def init_node():
+    from linked_list import Node
+    node = Node(0)
+    return str(type(node))
+
+@pytest.fixture
+def create_empty_node():
+    from linked_list import Node
+    return Node()
+
+@pytest.fixture
+def create_ll_with_list():
+    """Given a list, create a new Linked List with all the values."""
+    from linked_list import Linked_List
+    new_ll = Linked_List(PARAMS_SAMPLE_LIST)
+    return new_ll
+
+@pytest.fixture
+def create_empty_ll():
+    from linked_list import Linked_List
+    new_ll = Linked_List()
+    return new_ll
+
+
+def test_node_init():
+    """Test the initialization of a node."""
+    from linked_list import Node
+    assert type(Node) == type
+
+
+def test_node_instantiation(create_empty_node):
+    """Test if the Node objects gets initiated correctly"""
+    assert create_empty_node.value is None
+
+
+def test_node_value():
+    """Test the contents of nodes created with given input."""
+    from linked_list import Node
+    node2 = Node("Something")
+    assert node2.value == "Something" and node2.nxt == None
+
+
+def test_linked_list_init():
+    """Test initialization of the linked_list."""
+    from linked_list import Linked_List
+    assert type(Linked_List) == type
+
+
+def test_linked_list_push():
+    """Test the push method of the Linked List class."""
+    from linked_list import Linked_List
+    new_ll = Linked_List()
+    new_ll.push("Something")
+    assert new_ll.head.value == "Something" and new_ll._size == 1
+
+
+def test_linked_list_search_success():
+    """Test the search method of the Linked List class with a search query that exists."""
+    from linked_list import Linked_List
+    new_linked_list = Linked_List(PARAMS_SAMPLE_LIST)
+    result = new_linked_list.search('pear')
+    assert result.value == 'pear'
+
+
+def test_linked_list_search_failure():
+    """Test the search method of the Linked List class with a search query that DOES NOT exist."""
+    from linked_list import Linked_List
+    new_linked_list = Linked_List(PARAMS_SAMPLE_LIST)
+    result = new_linked_list.search('owt')
+    assert result.startswith('That')
+
+
+def test_linked_list_pop():
+    """Test the pop method of the Linked List class."""
+    from linked_list import Linked_List
+    new_ll = Linked_List(PARAMS_SAMPLE_LIST)
+    old_head_val = new_ll.head.value
+    new_ll.pop()
+    new_head_val = new_ll.head.value
+    assert old_head_val == "apple" and new_head_val == 3
```

Comment:
All this tests is that the pop method reassigns the head. It doesn't test that the `pop` method returns a value at all, let alone that it returns the value that was at `head`